RUMM 2326 Add expo plugin for mapping files (#233)
DataDog/dd-sdk-reactnative: docs/expo_crash_reporting.md

# Expo Crash Reporting and Error Tracking

## Overview

Enable Expo Crash Reporting and Error Tracking to get comprehensive crash reports and error trends with Real User Monitoring.

With this feature, you can access the following features:

-   Aggregated Expo crash dashboards and attributes
-   Symbolicated iOS and deobfuscated Android crash reports
-   Trend analysis with Expo error tracking

In order to symbolicate your stack traces and deobfuscate Android crashes, upload your .dSYM, Proguard mapping files and source maps to Datadog using the `expo-datadog` config plugin.

Your crash reports appear in [**Error Tracking**][1].

## Setup

Use the [`expo-datadog` package and configuration plugin][2]. For more information, see the [Expo and Expo Go documentation][3].

> Run `eas secret:create` to set `DATADOG_API_KEY` and `DD_API_KEY` to your Datadog API key.

Reviewer: We should propose to set only one of them, same change should be done in other places.

```suggestion
Run `eas secret:create` to set `DATADOG_API_KEY` or `DD_API_KEY` to your Datadog API key.
```

Author: I'll change it to `DATADOG_API_KEY` once the Gradle plugin accepts this environment variable. Until then we need both of them if we want to cover all mapping files.

### Disable file uploads

You can disable some files from uploading by setting the `iosDsyms`, `iosSourcemaps`, `androidProguardMappingFiles`, or `androidSourcemaps` parameters to `false`.

```json
{
    "expo": {
        "plugins": [
            [
                "expo-datadog",
                {
                    "iosDsyms": false
                }
            ]
        ]
    }
}
```

### Setting the Datadog site

> Run `eas secret:create` to set `DATADOG_SITE` to the host of your Datadog site, for example: `datadoghq.eu`. By default `datadoghq.com` will be used.

Reviewer 2: ```suggestion
Run `eas secret:create` to set `DATADOG_SITE` to the host of your Datadog site, for example: `datadoghq.eu`. By default, `datadoghq.com` is used.
```

> Due to the discrepancy in how your plugins are implemented, you also need to specify the `site` as a configuration plugin parameter, for example `EU1`. By default `US1` will be used.

Reviewer 2: ```suggestion
Due to the discrepancy in how your plugins are implemented, you need to specify the `site` as a configuration plugin parameter, for example `EU1`. By default, `US1` is used.
```

```json
{
    "expo": {
        "plugins": [
            [
                "expo-datadog",
                {
                    "site": "EU1"
                }
            ]
        ]
    }
}
```

### Plugin configuration options

> Due to the discrepancy in how your plugins are implemented, you need to specify the `site` as an environment secret and as a configuration plugin parameter.

Reviewer 2: ```suggestion
You also need to specify the `site` as an environment secret.
```
